solidity/test/TestPhasedEscrow.js

```diff
@@ -174,82 +198,294 @@ describe("PhasedEscrow", () => {
+        token.address,
+        rewardsContract.address,
+        {from: owner}
+      )
+      await rewardsBeneficiary.transferOwnership(phasedEscrow.address, {
+        from: owner,
+      })
+
+      await phasedEscrow.setBeneficiary(rewardsBeneficiary.address, {
+        from: owner,
+      })
+    })
+
+    assertRewards(baseBalance, transferAmount)
+  })
+
+  describe("when withdrawing to a ECDSARewardsEscrowBeneficiary", () => {
+    const baseBalance = 200000000
+    const transferAmount = 178200000
+
+    before(async () => {
+      rewardsContract = await TestECDSARewards.new(token.address)
+
+      rewardsBeneficiary = await ECDSARewardsEscrowBeneficiary.new(
+        token.address,
+        rewardsContract.address,
+        {from: owner}
+      )
+      await rewardsBeneficiary.transferOwnership(phasedEscrow.address, {
+        from: owner,
+      })
+
+      await phasedEscrow.setBeneficiary(rewardsBeneficiary.address, {
+        from: owner,
+      })
+    })
+
+    assertRewards(baseBalance, transferAmount)
+  })
+
+  describe("CurveRewardsEscrowBeneficiary", () => {
+    describe("__escrowSentTokens", async () => {
+      const baseBalance = 100000000
+      const transferAmount = 1000
+      let rewardsContract
+      let rewardsBeneficiary
 
       before(async () => {
-        curveRewards = await TestCurveRewards.new(token.address)
+        rewardsContract = await TestCurveRewards.new(token.address)
         rewardsBeneficiary = await CurveRewardsEscrowBeneficiary.new(
           token.address,
-          curveRewards.address
+          rewardsContract.address,
+          {from: owner}
         )
 
         const amount = web3.utils.toBN(baseBalance)
-        await token.transfer(phasedEscrow.address, amount, {from: owner})
+        await token.transfer(rewardsBeneficiary.address, amount, {from: owner})
+      })
 
-        await phasedEscrow.setBeneficiary(rewardsBeneficiary.address, {
+      it("can be called by the owner", async () => {
+        await rewardsBeneficiary.__escrowSentTokens(transferAmount, {
           from: owner,
         })
+        // ok, no revert
       })
 
-      beforeEach(createSnapshot)
-      afterEach(restoreSnapshot)
-
-      it("withdraws specified tokens from escrow", async () => {
-        await phasedEscrow.withdraw(transferAmount, {from: owner})
-
-        expect(await token.balanceOf(phasedEscrow.address)).to.eq.BN(
-          baseBalance - transferAmount,
-          "Unexpected amount withdrawn"
+      it("can not be called by the non-owner", async () => {
+        await expectRevert(
+          rewardsBeneficiary.__escrowSentTokens(transferAmount, {
+            from: otherAccount,
+          }),
+          "Ownable: caller is not the owner"
         )
       })
+    })
+  })
 
-      it("transfers specified tokens to curve rewards contract", async () => {
-        await phasedEscrow.withdraw(transferAmount, {from: owner})
+  describe("StakerRewardsBeneficiary", () => {
```

Comment:
Let's define them on the same level as `PhasedEscrow` as they are covering a separate contract:
```
describe("StakerRewardsBeneficiary", () => {
  const owner = accounts[0]
  const thirdParty = accounts[1]

  const transferAmount = 1000

  let token
  let rewardsContract
  let rewardsBeneficiary

  before(async () => {
    token = await KeepToken.new({from: owner})
    rewardsContract = await TestECDSARewards.new(token.address)
    rewardsBeneficiary = await StakerRewardsBeneficiary.new(
      token.address,
      rewardsContract.address,
      {from: owner}
    )

    const amount = web3.utils.toBN(transferAmount)
    await token.transfer(rewardsBeneficiary.address, amount, {from: owner})
  })

  describe("__escrowSentTokens", async () => {
    it("can be called by the owner", async () => {
      await rewardsBeneficiary.__escrowSentTokens(transferAmount, {
        from: owner,
      })
      // ok, no revert
    })

    it("can not be called by the non-owner", async () => {
      await expectRevert(
        rewardsBeneficiary.__escrowSentTokens(transferAmount, {
          from: thirdParty,
        }),
        "Ownable: caller is not the owner"
      )
    })
  })
})
```

Reply:
Right, done.